projects/RabbitMQ.Client/client/impl/ChannelBase.cs

```diff
@@ -783,13 +792,22 @@ protected void HandleConnectionUnblocked()
             Session.Connection.HandleConnectionUnblocked();
         }
 
-        protected void HandleQueueDeclareOk(in IncomingCommand cmd)
+        protected bool HandleQueueDeclareOk(in IncomingCommand cmd)
         {
-            var method = new Client.Framing.Impl.QueueDeclareOk(cmd.MethodBytes.Span);
-            cmd.ReturnMethodBuffer();
-            var k = (QueueDeclareRpcContinuation)_continuationQueue.Next();
-            k.m_result = new QueueDeclareOk(method._queue, method._messageCount, method._consumerCount);
-            k.HandleCommand(IncomingCommand.Empty); // release the continuation.
+            IRpcContinuation continuation = _continuationQueue.Peek();
+            if (continuation is QueueDeclareRpcContinuation)
+            {
+                var method = new Client.Framing.Impl.QueueDeclareOk(cmd.MethodBytes.Span);
+                cmd.ReturnMethodBuffer();
+                var k = (QueueDeclareRpcContinuation)_continuationQueue.Next();
+                k.m_result = new QueueDeclareOk(method._queue, method._messageCount, method._consumerCount);
+                k.HandleCommand(IncomingCommand.Empty); // release the continuation.
+                return true;
+            }
```

Comment:
```suggestion
            if(_continuationQueue.TryPeek<QueueDeclareRpcContinuation>(out var k))
            {
                _continuationQueue.Next()
                var method = new Client.Framing.Impl.QueueDeclareOk(cmd.MethodBytes.Span);
                cmd.ReturnMethodBuffer();
                k.m_result = new QueueDeclareOk(method._queue, method._messageCount, method._consumerCount);
                k.HandleCommand(IncomingCommand.Empty); // release the continuation.
                return true;
            }
```

Reply:
Might also make sense to also add a `TryPop<T>` or `TryNext<T>` which works like `TryPeek<T>` but also does the `_continuationQueue.Next()` call if the type matches.